### PR TITLE
Contrail microservices deploy prototype

### PR DESCRIPTION
**Pull request by pjrusak**
Review needed

### CODE REVIEW

File: roles/contrail-provision-dockers/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+- name: Debug hostvars 
+  debug:
+    var: hostvars[inventory_hostname]
+    verbosity: 4
```
Comment (kklimonda) [inline roles/contrail-provision-dockers/tasks/main.yaml]: This won't work in zuul - it never sets verbosity over 3 IIRC. Either remove verbosity, or the task.

File: roles/contrail-provision-dockers/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+- name: Clone contrail-ansible-deployer repository
+  git:
+    repo: 'https://github.com/Juniper/contrail-ansible-deployer'
+    version: 'a32639c61f4c5c20fddb99ccb25f75944cf2ca0a'
+    force: yes
```
Comment (kklimonda) [inline roles/contrail-provision-dockers/tasks/main.yaml]: No need to `force` anything - we always start with a blank slate.

File: roles/contrail-provision-dockers/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+- name: Debug hostvars 
+  debug:
+    var: hostvars[inventory_hostname]
+    verbosity: 4
+
+- name: Clone contrail-ansible-deployer repository
```
Comment (kklimonda) [inline roles/contrail-provision-dockers/tasks/main.yaml]: Before we make the job "final" we should probably move this to zuul merger

File: zuul.d/jobs.yaml [hunk 4]
```diff
@@ -108,3 +108,12 @@
       nodes:
         - name: builder
           label: ubuntu-xenial-builder
+
+- job:
+    name: contrail-vnc-docker-provision-centos74
+    voting: False
+    run: playbooks/docker/centos74-provision
```
Comment (kklimonda) [inline zuul.d/jobs.yaml]: This playbook should probably be `playbooks/contrail/centos74-provision-systest` or similar - this job will both provision and run tests.

File: zuul.d/jobs.yaml [hunk 5]
```diff
@@ -108,3 +108,12 @@
...
+    nodeset:
+      nodes:
+        - name: builder
+          label: centos-7.4-builder
```
Comment (kklimonda) [inline zuul.d/jobs.yaml]: This node "type" is probably way too large for provision/systest jobs (8vcpu/16GB sounds excessive)

File: zuul.d/jobs.yaml [hunk 6]
```diff
@@ -108,3 +108,12 @@
...
         - name: builder
           label: ubuntu-xenial-builder
+
+- job:
+    name: contrail-vnc-docker-provision-centos74
+    voting: False
+    run: playbooks/docker/centos74-provision
+    nodeset:
+      nodes:
+        - name: builder
```
Comment (kklimonda) [inline zuul.d/jobs.yaml]: This is no longer a "builder" but rather a "contrail-aio" (all-in-one) or something similar.

File: roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2 [hunk 7]
```diff
@@ -0,0 +1,16 @@
+---
+BUILD_VMS: false
+CONFIGURE_VMS: true
+CREATE_CONTAINERS: true
+CONTAINER_VM_CONFIG:
```
Comment (kklimonda) [inline roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2:5]: This seems to be configuration related to spawning a VM for containers? This is going to be done by nodepool and should probably not be here - we are not going to spawn VMs inside of another VMs.

**Review by kklimonda — CHANGES_REQUESTED** [review]

File: roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2 (hunk 7 above)
Comment (pjrusak) [reply roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2:5]: Right first variable controls spawning VMs so I disabled related task setting variable to false. Second is temporary on until nodepool image will be provided with dependencies.

File: zuul.d/jobs.yaml (hunk 5 above)
Comment (pjrusak) [reply zuul.d/jobs.yaml]: Do we have something more suitable right now?

File: roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2 (hunk 7 above)
Comment (kklimonda) [reply roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2:5]: What depenencies are those? Do you have a list somewhere?

Comment (pjrusak) [reply roles/contrail-provision-dockers/templates/inventory/group_vars/all.yaml.j2:5]: Yes, I have a list. It's a list matching following playbook: https://github.com/Juniper/contrail-ansible-deployer/blob/master/playbooks/roles/configure_container_hosts/tasks/install_software.yml

**Pull request merged**